src/components/FreeOffer/FreeOffer.js

```diff
@@ -28,25 +28,36 @@ const FreeOffer = ({ onPaymentComplete }) => {
 
   const offerType = offerId?.charAt(0);
   const icon = period || offerType;
+
   const generateDescriptionForFreeOffer = () => {
     switch (offerType) {
       case 'S': {
-        return `Free subscription`;
+        return t('free-offer.subscription', 'Free subscription');
       }
       case 'P': {
         if (!period) {
-          return `Access until ${dateFormat(expiresAt, true)}`;
+          return t('free-offer.pass', 'Access until {{date}}', {
+            date: dateFormat(expiresAt, true)
+          });
         }
-        return `${periodMapper[period].accessText} free pass`;
+        return `${t(
+          `period.${periodMapper[period].accessText.toLowerCase()}`,
+          periodMapper[period].accessText
+        )} ${t('free-offer.free-pass', 'free pass')}`;
```

Comment:
This is quest complex and hard to read. Also, it will be difficult to translate when languages will require a different word order. Could we split it into one translation with multiple periods, eg.
`free-offer.free-pass.daily`: 'Daily pass',
`free-offer.free-pass.weekly`: 'Weekly pass',
....
`free-offer.free-pass.annual`: 'Yearly pass',
`free-offer.free-pass.date`: 'Access until',
?

Reply:
But then we would have combination of every period + `free pass,` `free access`, `season pass`, `access`, so almost 40 translation keys